first-party/jib-layer-filter-extension-maven/src/main/java/com/google/cloud/tools/jib/maven/extension/layerfilter/JibLayerFilterExtension.java

```diff
@@ -102,9 +122,120 @@ public ContainerBuildPlan extendContainerBuildPlan(
+
+    ContainerBuildPlan newPlan = newPlanBuilder.build();
+
+    if (config.get().isCreateParentLayers()) {
+      newPlan = moveParentDepsToNewLayers(newPlan, mavenData, logger);
+    }
+
+    return newPlan;
+  }
+
+  private ContainerBuildPlan moveParentDepsToNewLayers(
+      ContainerBuildPlan buildPlan, MavenData mavenData, ExtensionLogger logger)
+      throws JibPluginExtensionException {
+
+    logger.log(LogLevel.LIFECYCLE, "Moving parent dependencies to new layers.");
+
+    // the key is the expected path for the parent dependency
+    Map<String, Artifact> parentDependencies =
+        getParentDependencies(mavenData)
+            .stream()
+            .map(d -> d.getArtifact())
+            // TODO: configurable?
+            .collect(
+                // TODO: getVersion or getBaseVersion? check in jib, which version it is...
+                Collectors.toMap(
+                    a -> "/app/libs/" + a.getArtifactId() + "-" + a.getVersion() + ".jar", a -> a));
+
+    // parent dependencies that have not been found in any layer (due to different version or
+    // filtering)
+    Map<String, Artifact> parentDependenciesNotFound = new HashMap<>(parentDependencies);
+
+    List<FileEntriesLayer.Builder> newLayers = new ArrayList<>();
+
+    @SuppressWarnings("unchecked")
+    List<FileEntriesLayer> originalLayers = ((List<FileEntriesLayer>) buildPlan.getLayers());
+    originalLayers.forEach(
+        l -> {
+          // for each layer, create a parent layer
+          String parentLayerName = l.getName() + "-parent";
+          FileEntriesLayer.Builder toParentLayerBuilder =
+              FileEntriesLayer.builder().setName(parentLayerName);
+          newLayers.add(toParentLayerBuilder);
+          // ... and the normal layer
+          FileEntriesLayer.Builder toLayerBuilder = FileEntriesLayer.builder().setName(l.getName());
+          newLayers.add(toLayerBuilder);
+
+          l.getEntries()
+              .forEach(
+                  fe -> {
+                    String path = fe.getExtractionPath().toString();
+                    if (parentDependencies.containsKey(path)) {
+                      // move to parent layer
+                      logger.log(LogLevel.INFO, "Moving " + path + " to " + parentLayerName + ".");
+                      toParentLayerBuilder.addEntry(fe);
+                      // mark parent dep as found
+                      parentDependenciesNotFound.remove(path);
+                    } else {
+                      // keep in original layer
+                      logger.log(LogLevel.INFO, "Keep " + path + " in " + l.getName() + ".");
+                      toLayerBuilder.addEntry(fe);
+                    }
+                  });
+        });
+
+    parentDependenciesNotFound.forEach(
+        (path, artifact) -> {
+          logger.log(LogLevel.ERROR, "Dependency from parent not found: " + path);
+        });
+
+    return buildPlanWithNewLayers(buildPlan, newLayers);
```

Comment:
How about 
```java
    // Return a modified build plan with the new layers.
    List<FileEntriesLayer> newLayers =
        newLayerBuilders
            .stream()
            .map(FileEntriesLayer.Builder::build)
            .filter(layer -> !layer.getEntries().isEmpty())
            .collect(Collectors.toList());
    return buildPlan.toBuilder().setLayers(newLayers).build();
```
without the method after renaming `newLayers` to `newLayerBuilders`?

Reply:
ok